cmd/nodepool/azure/create.go

```diff
@@ -2,69 +2,183 @@ package azure
+	flags.StringVar(&opts.MarketplaceSKU, "marketplace-sku", opts.MarketplaceSKU, "The Azure Marketplace image SKU.")
+	flags.StringVar(&opts.MarketplaceVersion, "marketplace-version", opts.MarketplaceVersion, "The Azure Marketplace image version.")
+	flags.BoolVar(&opts.MarketplaceThirdPartyImage, "marketplace-third-party-image", opts.MarketplaceThirdPartyImage, "The Azure Marketplace image flag indicating if this is a third party image.")
+}
+
+func BindDeveloperOptions(opts *RawAzurePlatformCreateOptions, flags *pflag.FlagSet) {
+	bindCoreOptions(opts, flags)
+}
+
+// validatedAzurePlatformCreateOptions is a private wrapper that enforces a call of Validate() before Complete() can be invoked.
+type validatedAzurePlatformCreateOptions struct {
+	*RawAzurePlatformCreateOptions
+}
+
+type ValidatedAzurePlatformCreateOptions struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*validatedAzurePlatformCreateOptions
+}
+
+// completedCreateOptions is a private wrapper that enforces a call of Complete() before nodepool creation can be invoked.
+type completetedAzurePlatformCreateOptions struct {
+	*AzurePlatformCreateOptions
+	*AzureMarketPlaceImageInfo
+}
+
+type CompletedAzurePlatformCreateOptions struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*completetedAzurePlatformCreateOptions
+}
+
+func (o *RawAzurePlatformCreateOptions) Validate() (*ValidatedAzurePlatformCreateOptions, error) {
+	// Validate publisher, offer, sku, and version flags are provided if one of them is not empty
+	marketplaceImageInfo := map[string]*string{
+		"marketplace-publisher": &o.MarketplacePublisher,
+		"marketplace-offer":     &o.MarketplaceOffer,
+		"marketplace-sku":       &o.MarketplaceSKU,
+		"marketplace-version":   &o.MarketplaceVersion,
 	}
+	if err := util.ValidateMarketplaceFlags(marketplaceImageInfo); err != nil {
+		return nil, err
+	}
+
+	return &ValidatedAzurePlatformCreateOptions{
+		validatedAzurePlatformCreateOptions: &validatedAzurePlatformCreateOptions{
+			RawAzurePlatformCreateOptions: o,
+		},
+	}, nil
+}
+
+func (o *ValidatedAzurePlatformCreateOptions) Complete() (*CompletedAzurePlatformCreateOptions, error) {
+	return &CompletedAzurePlatformCreateOptions{
+		completetedAzurePlatformCreateOptions: &completetedAzurePlatformCreateOptions{
+			AzurePlatformCreateOptions: o.AzurePlatformCreateOptions,
+			AzureMarketPlaceImageInfo:  o.AzureMarketPlaceImageInfo,
+		},
+	}, nil
+}
 
+func NewCreateCommand(coreOpts *core.CreateNodePoolOptions) *cobra.Command {
+	platformOpts := DefaultOptions()
 	cmd := &cobra.Command{
 		Use:          "azure",
 		Short:        "Creates basic functional NodePool resources for Azure platform",
 		SilenceUsage: true,
 	}
+	BindDeveloperOptions(platformOpts, cmd.Flags())
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		validOpts, err := platformOpts.Validate()
+		if err != nil {
+			return err
```

Comment:
can we have test Data fixture for this same than we have for cluster create?

Reply:
We have one for the new marketplace image unless you are referring to something else? cmd/cluster/azure/testdata/zz_fixture_TestCreateCluster_create_with_a_ure_marketplace_image.yaml